### PR TITLE
Typescript 2.0 and the Module Resolution

### PR DESCRIPTION
**Pull request by wassertim**
This PR updates the typescript version to the newest 2.0 and fixes the module resolution, so the baseUrl is set by default and points to the webjars/lib directory. No npm install is needed anymore before compiling the project with sbt.

I also added traceModuleResolution (so we can set this to true and see what went wrong in a compilation).

When I wrote the test I used `"org.webjars.npm" % "angular__core" % "2.0.0-rc.4"` for testing the module resolution. This webjar is extracted to the lib/angular__core, but the official angular directory in the modules should be lib/@angular/core. I added the paths option for it so we can map paths now, i.e. "@angular/core" to "angular__core.

It fixes the issues: https://github.com/ArpNetworking/sbt-typescript/issues/31 and https://github.com/ArpNetworking/sbt-typescript/issues/37
@BrandonArp

### CODE REVIEW

File: src/sbt-test/sbt-typescript-plugin/module-resolution/app/assets/javascripts/es6-shim.d.ts [hunk 1]
```diff
@@ -0,0 +1,670 @@
+// Compiled using typings@0.6.10
```
Comment (wassertim) [inline src/sbt-test/sbt-typescript-plugin/module-resolution/app/assets/javascripts/es6-shim.d.ts:1]: This file is needed only for compilation and should be there on a plugin's user side too. Ignore it while reviewing.

**Comment by wassertim** [comment]
The build failed because of the old JDK. The play version which is used in the test requires the JDK 8.
@BrandonArp when are we going to update JDK in Travis? ;)

I will try though to use a previous version of the play-framework in the test in order to fix the continuous integration

**Comment by bjoernhaeuser** [comment]
Looks awesome!

**Comment by wassertim** [comment]
I dropped the old JDK instead. It works now :)

**Comment by BrandonArp** [comment]
Very nice.  You're good to merge after a rebase.

**Comment by wassertim** [comment]
@BrandonArp Will you publish a new version?

**Comment by blast-hardcheese** [comment]
@wassertim: Do you have a full example or documentation about how to use `paths`? I see the paths you're talking about in [org.webjars.npm/angular_core/2.0.0-rc.4](http://www.webjars.org/listfiles/org.webjars.npm/angular__core/2.0.0-rc.4), but it doesn't seem like [org.webjars.npm/react/15.3.0](http://www.webjars.org/listfiles/org.webjars.npm/react/15.3.0) has this particular problem, but I'm still getting

```
[error] .../src/main/assets/app.tsx:0: Cannot find module 'react'.
[error] import * as React from "react"
```

**Comment by blast-hardcheese** [comment]
Actually, I think I might be running into a different issue -- When looking at `target/web/web-modules/main/webjars/lib`, there's only a `types__react/react/` (I depend on both `"org.webjars.npm" % "react" % 15.3.0` and `"org.webjars.npm" % "types_react" % 0.14.29`). I'm wondering if the `react` webjar not being unpacked is what's really going wrong here.

**Comment by blast-hardcheese** [comment]
My `build.sbt` contains:

```
TypescriptKeys.jsx := "React"

TypescriptKeys.traceResolution := true

TypescriptKeys.paths := Map(
  "react" -> List("types__react/react")
)

lazy val root = project.in(file(".")).enablePlugins(SbtWeb)
```

which now (fortunately!) compiles, although I suspect I might still be doing something wrong here.

**Pull request merged**